### ISSUE TITLE
TChain::CopyAddresses:0 warning in RDataFrame Snapshot of TChain with multiple files with objects

### ISSUE THREAD

**Issue by dudarboh** [root-project/root#7727]
- [x] Checked for duplicates

### Describe the bug
1. Use `TChain` with >1 root files as an input for RDataFrame
2. TChain should have objects like `XYZVector` or `vector <double>`, etc.
3. Snapshot RDataFrame providing list of columns
4. Warning `TChain::CopyAddresses:0: RuntimeWarning: Could not find branch named 'xxx' in tree named 'xxx'`appears
5. Warning appears on the transition between files.

### Expected behavior
Expected no warning if it has no effect on the output.
Error, if it effects the output.
List of possible scenarios  when it cause code to break?

### To Reproduce
Compact reproducer:

```
import ROOT

ROOT.gInterpreter.Declare(
'''
#include "Math/Vector3D.h"
using namespace ROOT::Math;
''')

def test_warning():
    # Create two files to read data from
    # Warning appears only for cpp_vec and phys_vec branches
    for i in range(2):
        ROOT.RDataFrame(10).Define("simple_number", "rdfentry_")\
                           .Define("cpp_vec", "std::vector<double>{-1, 42, -1}")\
                           .Define("phys_vec", "XYZVector(-1, 42, -1)")\
                           .Snapshot("test_tree", "file{}.root".format(i))

    ch = ROOT.TChain("test_tree")
 
    # Using only one file shows No warning...
    # ch.Add("./file1.root")
    ch.Add("./file[01].root")

    df = ROOT.RDataFrame(ch).Filter('{cout<<"Event: "<<rdfentry_<< endl;} return true;')
    df = df.Define("new_var", "phys_vec.Y()*cpp_vec[1]")

    # Snapshoting w/o branch list shows no warning too
    # df.Snapshot("new_tree", "new_file.root")
    df.Snapshot("new_tree", "new_file.root", ["new_var"])


test_warning()
```

### Setup
1. ROOT 6.22/00 from  /cvmfs/sft.cern.ch/lcg/views/LCG_98python3/x86_64-centos7-gcc10-opt/bin/root
2. Centos 7
4. Python 3.7.6

### Additional Information

This behaviour was spotted and discussed at ROOT forum:
https://root-forum.cern.ch/t/warning-in-tchain-copyaddresses-could-not-find-branch-named/35055
https://root-forum.cern.ch/t/tchain-0-warning-with-rdataframe-could-not-find-branch-named/44186

**Comment by eguiraud** [comment]
Hi @dudarboh , thank you for the report, I will take a look as soon as possible (i.e. after Easter I'm afraid :sweat_smile: ).

**Comment by eguiraud** [comment]
This issue seems related to https://sft.its.cern.ch/jira/browse/ROOT-9487 .

**Comment by eguiraud** [comment]
Ok yes, this is exactly the same underlying issue as ROOT-9487 mentioned above, but for single-thread runs this time, so the workaround mentioned in ROOT-9487 does not work (and indeed adding `ROOT.EnableImplicitMT()` at the top of the repro gets rid of the warnings).

The issue: `Snapshot` adds the output tree as a clone of the input chain, because the output tree needs to be informed when the addresses of the input branches change (as it happens when the chain switches from one tree to the next). Now, when performing the address updates, the input chain might complain _once per active branch_ if it does not find a corresponding branch in the output tree.

I don't know how to solve this elegantly. The only idea I have is an ad-hoc temporary silencing of these warnings at the point at which an input chain in RDF switches files.

@pcanal might have a better proposal.

**Comment by pcanal** [comment]
One thing RDF could do is:
```
chain->SetBranchStatus("*", kFALSE);
for( name : list_of_names_of_used_top_level_used) 
   chain->SetBranchStatus(std::string(name ) + "*", kTRUE);
```
This would have the side effect to have all the TTree interface ignore the unused branches (that includes not copying them as part of CloneTree).

**Comment by eguiraud** [comment]
This should be fixed as a side-effect of #8375 if we manage to actually merge it.

**Comment by eguiraud** [comment]
I just ran the repro in the issue description against #8375 and I can confirm it fixes it.
